Add trace points for injected bundle and user script evaluation

bnham opened the pull request:
#### 06b5d3e6bd28859e13b589b1bc101905d6fdedaa
<pre>
Add trace points for injected bundle and user script evaluation
<a href="https://bugs.webkit.org/show_bug.cgi?id=299878">https://bugs.webkit.org/show_bug.cgi?id=299878</a>
<a href="https://rdar.apple.com/161662816">rdar://161662816</a>

Reviewed by Keith Miller.

We are moving more functionality from the injected bundle to WKUserScript. To help with analyzing
perf traces, we need to add relevant signposts:

- The injected bundle mainly injects scripts via JSScriptRef, so I added signposts around creating
  and evaluating JSScriptRef.

- We already added some user script signposts in 300184@main, but since user scripts typically carry
  no URL, it can be hard to figure out what script is being evaluated. Log the first line of the
  script in the signpost to help with this (up to a limit of 64 characters).

The signpost macros are guarded by a `kdebug_is_enabled` check so we don&apos;t do any of the work
related to the signposts (e.g. getting the debugDescription) unless signposts are actually enabled.

* Source/JavaScriptCore/API/JSScriptRef.cpp:
* Source/WTF/wtf/SystemTracing.h:
* Source/WebCore/page/LocalFrame.cpp:
(WebCore::LocalFrame::injectUserScriptImmediately):
* Source/WebCore/page/UserScript.cpp:
(WebCore::UserScript::debugDescription const):
* Source/WebCore/page/UserScript.h:

Canonical link: <a href="https://commits.webkit.org/300772@main">https://commits.webkit.org/300772@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e64acb04fc204a8df81d82bf181e28028d3ad4f5

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/123799 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/43514 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/34211 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/130578 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/75949 "Built successfully") | ⏳ 🛠 ios-apple 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/125676 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/44238 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/52108 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/94150 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/62480 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/7e3dbdd5-cd9c-4d53-883a-8d9a2c730076) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/126752 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/35249 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/110750 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/74751 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/89e123c5-9f18-4c74-93e5-76091cba593d) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/34205 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/28910 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/74061 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/115958 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/104973 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/29133 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/133273 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/122331 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/50751 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/38649 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/102630 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/51126 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/106969 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/102459 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26046 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/47799 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/26054 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/47601 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/50605 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/56369 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/153427 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/50079 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/39001 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/53425 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/51753 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

kmiller68 reviewed (approved):
r=me. Although as a nit, you don't need the list of changed files/functions if you didn't add any comments.

bnham committed:
Add trace points for injected bundle and user script evaluation
https://bugs.webkit.org/show_bug.cgi?id=299878
rdar://161662816

Reviewed by Keith Miller.

We are moving more functionality from the injected bundle to WKUserScript. To help with analyzing
perf traces, we need to add relevant signposts:

- The injected bundle mainly injects scripts via JSScriptRef, so I added signposts around creating
  and evaluating JSScriptRef.

- We already added some user script signposts in 300184@main, but since user scripts typically carry
  no URL, it can be hard to figure out what script is being evaluated. Log the first line of the
  script in the signpost to help with this (up to a limit of 64 characters).

The signpost macros are guarded by a `kdebug_is_enabled` check so we don't do any of the work
related to the signposts (e.g. getting the debugDescription) unless signposts are actually enabled.

* Source/JavaScriptCore/API/JSScriptRef.cpp:
* Source/WTF/wtf/SystemTracing.h:
* Source/WebCore/page/LocalFrame.cpp:
(WebCore::LocalFrame::injectUserScriptImmediately):
* Source/WebCore/page/UserScript.cpp:
(WebCore::UserScript::debugDescription const):
* Source/WebCore/page/UserScript.h:

Canonical link: https://commits.webkit.org/300772@main
